service/pipeline.ml

```diff
@@ -21,20 +21,47 @@ let platforms =
 (* Link for GitHub statuses. *)
 let url ~owner ~name ~hash = Uri.of_string (Printf.sprintf "https://ci.ocamllabs.io/github/%s/%s/commit/%s" owner name hash)
 
+(* Link for GitHub CheckRun details. *)
+let url_variant ~owner ~name ~hash ~variant =
+  Printf.sprintf "https://ci.ocamllabs.io/github/%s/%s/commit/%s/variant/%s" owner name hash variant
+
 let opam_repository_commit =
   let repo = { Github.Repo_id.owner = "ocaml"; name = "opam-repository" } in
   Github.Api.Anonymous.head_of repo @@ `Ref "refs/heads/master"
 
-let github_status_of_state ~head result =
+let github_status_of_state ~head result results =
   let+ head = head
-  and+ result = result in
+  and+ result = result
+  and+ results = results in
   let { Github.Repo_id.owner; name } = Github.Api.Commit.repo_id head in
   let hash = Github.Api.Commit.hash head in
   let url = url ~owner ~name ~hash in
+  let variant_compare variant1 variant2 = match (variant1, variant2) with
```

Comment:
I don't think you need this. The job label should be `(analysis)`, which will already sort first because `(` comes before the letters.